client/my-sites/themes/sheet.jsx

```diff
@@ -134,6 +144,32 @@ const ThemeSheet = React.createClass( {
 		);
 	},
 
+	renderDownload() {
+		var downloadText;
+		if ( 'Free' !== this.props.price ) {
+			return '';
+		}
+		downloadText = this.props.download
+		? i18n.translate( 'This theme is available for download to be used on your {{u}}WordPress self-hosted{{/u}} installation.', {
```

Comment:
Yep, should be something specific, but initially the wordpress.org site is the starting point. :)